### PR TITLE
Fix/status update trigger

### PR DESCRIPTION
**Pull request by SteKoe**

### CODE REVIEW

File: spring-boot-admin-server/src/main/java/de/codecentric/boot/admin/server/services/AbstractEventHandler.java [hunk 1]
```diff
@@ -58,7 +57,7 @@ public void start() {
 			.ofType(this.eventType)
 			.cast(this.eventType)
 			.transform(this::handle)
-			.retryWhen(Retry.indefinitely().doBeforeRetry((s) -> this.log.warn("Unexpected error", s.failure())))
+			.onErrorContinue((throwable, o) -> this.log.warn("Unexpected error", throwable))
```
Comment (SteKoe) [inline spring-boot-admin-server/src/main/java/de/codecentric/boot/admin/server/services/AbstractEventHandler.java:60]: `onErrorContinue` _drops_ the error, logs it but will keep the subscription running.

File: spring-boot-admin-server/src/main/java/de/codecentric/boot/admin/server/services/IntervalCheck.java [hunk 2]
```diff
@@ -52,6 +52,7 @@ public class IntervalCheck {
 
 	private final Function<InstanceId, Mono<Void>> checkFn;
 
+	@lombok.Getter
```
Comment (SteKoe) [inline spring-boot-admin-server/src/main/java/de/codecentric/boot/admin/server/services/IntervalCheck.java:55]: Add getter to allow access from `StatusUpdateTrigger` to set timeout on interval check. This prevents stream overflows.

File: spring-boot-admin-server/src/main/java/de/codecentric/boot/admin/server/services/StatusUpdater.java [hunk 3]
```diff
@@ -59,16 +60,22 @@ public class StatusUpdater {
 
 	private final ApiMediaTypeHandler apiMediaTypeHandler;
 
+	private Duration timeout = Duration.ofSeconds(10);
```
Comment (SteKoe) [inline spring-boot-admin-server/src/main/java/de/codecentric/boot/admin/server/services/StatusUpdater.java:63]: Default timeout is 10 seconds as the default interval check in `IntervalCheck` is 10 seconds as well. Can be overridden by `timeout` method.

File: spring-boot-admin-server/src/main/java/de/codecentric/boot/admin/server/services/StatusUpdater.java [hunk 4]
```diff
@@ -82,11 +89,20 @@ protected Mono<Instance> doUpdateStatus(Instance instance) {
 			.uri(Endpoint.HEALTH)
 			.exchangeToMono(this::convertStatusInfo)
 			.log(log.getName(), Level.FINEST)
+			.timeout(getTimeoutWithMargin())
```
Comment (SteKoe) [inline spring-boot-admin-server/src/main/java/de/codecentric/boot/admin/server/services/StatusUpdater.java:92]: Apply the timeout here, as well.

File: spring-boot-admin-server/src/main/java/de/codecentric/boot/admin/server/services/StatusUpdater.java [hunk 5]
```diff
@@ -46,10 +47,10 @@
  *
  * @author Johannes Edmeier
  */
+@Slf4j
+@RequiredArgsConstructor
```
Comment (ulischulte) [inline spring-boot-admin-server/src/main/java/de/codecentric/boot/admin/server/services/StatusUpdater.java:51]: Just some minor refactorings...

**Review by ulischulte — APPROVED** [review]
LGTM

**Review by ulischulte — APPROVED** [review]

**Pull request merged**